src/Input/Select/Select.tsx

```diff
@@ -61,6 +55,38 @@ class Select extends React.Component<Props, State> {
     document.addEventListener('click', this.handleOnBlur, false);
   }
 
+  componentDidUpdate(prevProps: Props) {
+    const hasDifferentChildren = !isEqual(this.props.children,  prevProps.children);
+    if (hasDifferentChildren) {
+      this.setState({
+        selectedValue: this.props.value as string,
+        defaultValue: this.props.value as string,
+        floating: !!this.props.value,
```

Comment:
Again, more of personal preference but I'd prefer to make this more verbose by using `Boolean()` instead of the double bang `!!`. Some may argue that it is less performant but it is probably quite negligible so I'd go for more readable code :smile:

Reply:
got it, updated!